clients/registry-amqp/src/main/java/org/eclipse/hono/client/registry/amqp/ProtonBasedCredentialsClient.java

```diff
@@ -93,38 +87,22 @@ public ProtonBasedCredentialsClient(
         connection.getVertx().eventBus().consumer(
                 Constants.EVENT_BUS_ADDRESS_TENANT_TIMED_OUT,
                 this::handleTenantTimeout);
-        this.notificationReceiver = Objects.requireNonNull(notificationReceiver);
 
         if (isCachingEnabled()) {
-            notificationReceiver.registerConsumer(AllDevicesOfTenantDeletedNotification.TYPE,
+            NotificationEventBusSupport.registerConsumer(connection.getVertx(), AllDevicesOfTenantDeletedNotification.TYPE,
```

Comment:
I have to admit that I do not really like the tight coupling to the event bus here. Couldn't we instead inject a NotificationReceiver instance into the constructor which reads the notifications from the event bus?

Reply:
I actually see this as an explicit advantage of the solution here, that there is no need to pass around `NotificationReceiver` instances anymore (and `NotificationSender` objects in a follow-up PR). In that sense it's similar to the way the tenant-timeout mechanism is implemented - we also don't pass around TenantTimeoutReceiver objects for that.

Adding a `NotificationReceiver` here would mean that we need to introduce 
- a `MessagingSystemNotificationReceiver` (or something like that)
- and a `LocalViaEventBusNotificationReceiver` (or something like that).
Compared to that, I think the EventBus solution here is more straightforward and the leaner solution.

Regarding the coupling to the event bus here, I don't really see an issue with that.
When unit-testing this feature, one has to use a real (not mocked) `Vertx` instance now, but this is no big deal FMPOV (it will test the whole notification-via-event-bus sending process then).

Reply:
ok, let's see how we fare with this approach